code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -234,4 +234,157 @@ proc/flame_radius(radius = 1, turf/T, burn_intensity = 25, burn_duration = 25, b
+	icon_state = "grenade_impact"
+	item_state = "grenade_impact"
+	hud_state = "grenade_frag"
+	det_time = 40
+	dangerous = TRUE
+	underslug_launchable = TRUE
+
+/obj/item/explosive/grenade/impact/prime()
+	spawn(0)
+		explosion(loc, -1, -1, 1, 2)
+		qdel(src)
+	return
+
+/obj/item/explosive/grenade/impact/flamer_fire_act()
+	var/turf/T = loc
+	qdel(src)
+	explosion(T, -1, -1, 1, 2)
+
+/obj/item/explosive/grenade/impact/throw_impact(atom/hit_atom, speed)
+	. = ..()
+	if(launched && active && !istype(hit_atom, /turf/open)) //Only contact det if active, we actually hit something, and we're fired from a grenade launcher.
+		explosion(loc, -1, -1, 1, 2)
+		qdel(src)
+
+
+/obj/item/explosive/grenade/flare
+	name = "\improper M40 FLDP grenade"
+	desc = "A TGMC standard issue flare utilizing the standard DP canister chassis. Capable of being loaded in the M92 Launcher, or thrown by hand."
+	icon_state = "flare_grenade"
+	det_time = 0
+	throwforce = 1
+	dangerous = FALSE
+	underslug_launchable = TRUE
+	w_class = 2
+	hud_state = "grenade_frag"
+	var/fuel = 0
+
+/obj/item/explosive/grenade/flare/New()
+	fuel = rand(800, 1000) // Sorry for changing this so much but I keep under-estimating how long X number of ticks last in seconds.
+	return ..()
+
+/obj/item/explosive/grenade/flare/flamer_fire_act()
+	if(!active)
+		turn_on()
+
+/obj/item/explosive/grenade/flare/Destroy()
+	if(ismob(loc))
+		loc.SetLuminosity(-FLARE_BRIGHTNESS)
+	else
+		SetLuminosity(0)
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/explosive/grenade/flare/process()
+	fuel = max(fuel - 1, 0)
+	if(!fuel || !active)
+		turn_off()
+		if(!fuel)
+			icon_state = "[initial(icon_state)]-empty"
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/explosive/grenade/flare/proc/turn_off()
+	active = FALSE
+	heat_source = 0
+	force = initial(force)
+	damtype = initial(damtype)
+	if(ismob(loc))
+		var/mob/U = loc
+		update_brightness(U)
+	else
+		update_brightness(null)
+
+/obj/item/explosive/grenade/flare/proc/turn_on()
+	active = TRUE
+	force = FLARE_IGNITE_MELEE_DAMAGE
+	throwforce = FLARE_IGNITE_THROW_DAMAGE
+	igniting = TRUE
+	item_fire_stacks = FLARE_IGNITE_STACKS
+	heat_source = 1500
+	damtype = "fire"
+	update_brightness()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/explosive/grenade/flare/attack_self(mob/user)
+
+	// Usual checks
+	if(!fuel)
+		to_chat(user, "<span class='notice'>It's out of fuel.</span>")
+		return
+	if(active)
+		return
+
+	// All good, turn it on.
+	user.visible_message("<span class='notice'>[user] activates the flare.</span>", "<span class='notice'>You depress the ignition button, activating it!</span>")
+	turn_on(user)
+
+/obj/item/explosive/grenade/flare/activate(mob/user)
+	if(!active)
+		turn_on(user)
+
+/obj/item/explosive/grenade/flare/on/New()
+
+	. = ..()
+	active = TRUE
+	heat_source = 1500
+	update_brightness()
+	force = FLARE_IGNITE_MELEE_DAMAGE
```

Comment:
This doesn't need a define. It's a straight number, not a magical one. Adding a define here only obscures it.